core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -1206,6 +1236,563 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
+                            PUSH_I64(end_addr);
+                            break;
+                        case WASM_OP_DELEGATE:
+                            skip_leb(lookup_cursor); /* skip depth */
+                            _EXCEVERBOSE(
+                                "try: WASM_OP_DELEGATE catchhandler %p",
+                                end_addr);
+                            PUSH_I64(end_addr);
+                            /* patch target_addr */
+                            (frame_csp - 1)->target_addr = lookup_cursor;
+                            break;
+                        case WASM_OP_END:
+                            _EXCEVERBOSE("try: WASM_OP_END");
+                            PUSH_I64(0);
+                            /* patch target_addr */
+                            (frame_csp - 1)->target_addr = end_addr;
+                            break;
+                        default:
+                            /* something went wrong */
+                            wasm_set_exception(module,
+                                               "find block address returned an "
+                                               "unexpected opcode");
+                            goto got_exception;
+                    }
+                    /* ... search until the returned address is the END of the
+                     * TRY block */
+                } while (handler_opcode != WASM_OP_END
+                         && handler_opcode != WASM_OP_DELEGATE);
+                /* handler setup on stack complete */
+
+                HANDLE_OP_END();
+            }
+            HANDLE_OP(WASM_OP_CATCH)
+            {
+                _EXCEVERBOSE(
+                    "hit a catch clause, leave the frame, target_addr %p\n",
+                    (frame_csp - 1)->target_addr);
+
+                /* skip the tag_index */
+                skip_leb(frame_ip);
+                /* leave the frame */
+                POP_CSP_N(0);
+                HANDLE_OP_END();
+            }
+            HANDLE_OP(WASM_OP_CATCH_ALL)
+            {
+                _EXCEVERBOSE("hit a catch_all clause, leave the frame\n");
+                /* leave the frame */
+                POP_CSP_N(0);
+                HANDLE_OP_END();
+            }
+            HANDLE_OP(WASM_OP_DELEGATE)
+            {
+                _EXCEVERBOSE(
+                    "hit a delegate clause, leave the frame, target_addr %p\n",
+                    (frame_csp - 1)->target_addr);
+                /* skip the delegate depth */
+                skip_leb(frame_ip);
+                /* leave the frame like WASM_OP_END */
+                POP_CSP();
+                HANDLE_OP_END();
+            }
+#else
+            HANDLE_OP(WASM_OP_UNUSED_0x06)
+            HANDLE_OP(WASM_OP_UNUSED_0x07)
+            HANDLE_OP(WASM_OP_UNUSED_0x08)
+            HANDLE_OP(WASM_OP_UNUSED_0x09)
+            HANDLE_OP(WASM_OP_UNUSED_0x18)
+            HANDLE_OP(WASM_OP_UNUSED_0x19)
```

Comment:
Had better change `WASM_OP_UNUSED_0x06/0x07/...` to `WASM_OP_TRY/CATCH/...` in wasm_opcode.h, and remove the else branch here, and keep the original code: L3819 to L3826 in the original file.

Reply:
agreed; code updated accordingly.

Reply:
Updated accordingly